File: SIC_Backend/Data/Models/Place.cs
namespace SIC_Backend.Data.Models
{
    public class Place
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public required string UserId { get; set; }

        public User? User { get; set; }
        public ICollection<Section> Sections { get; set; } = [];
        public ICollection<Device> Devices { get; set; } = [];

        public static Place FromCreateModel(CreatePlaceModel model)
        {
            return new Place
            {
                Name = model.Name,
                UserId = model.UserId
            };
        }
    }
}

File: sic_home/lib/models/place.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sic_home/models/section.dart';

part 'place.g.dart';

@JsonSerializable()
class Place {
  int id;
  String name;
  String userId;
  List<Section> sections;

  Place({
    required this.id,
    required this.name,
    required this.userId,
    required this.sections,
  });

  factory Place.fromJson(Map<String, dynamic> json) => _$PlaceFromJson(json);

  Map<String, dynamic> toJson() => _$PlaceToJson(this);
}
